### PR TITLE
Made usage of strimzi entity operator optional.

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

The strimzi settings in the cloudflow-operator are now optional. If these are not provided, the cloudflow operator will not create Strimzi KafkaTopic CRs, which are used by the strimzi topic operator.

### Why are the changes needed?

Users have asked if it is possible to install and run Cloudflow without strimzi kafka cluster and topic operator.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

all green, tested on gke.

What still has to happen (in a separate PR) is that the helm chart for installing cloudflow should make it possible to configure the default kafka broker that the cloudflow operator will use for managed topics.

### CODE REVIEW

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Actions.scala [hunk 1]
```diff
@@ -93,7 +93,20 @@ object Actions {
       currentApp: Option[CloudflowApplication.CR],
       deleteOutdatedTopics: Boolean = false
   )(implicit ctx: DeploymentContext): Seq[Action[ObjectResource]] =
-    TopicActions(newApp, currentApp, deleteOutdatedTopics)
+    if (ctx.kafkaContext.strimziTopicOperatorNamespace.nonEmpty && ctx.kafkaContext.strimziClusterName.nonEmpty) {
```
Comment (ennru) [inline core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Actions.scala:96]: Shouldn't this be a `useStrimzi` method on `KafkaContext`?

File: core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala [hunk 2]
```diff
@@ -22,8 +22,8 @@ trait TestDeploymentContext {
   implicit val ctx: DeploymentContext =
```
Comment (ennru) [inline core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala:22]: Should this get a better name to express it relies on Strimzi?

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Actions.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Actions.scala:96]: yeah could do that. This is really a quick fix..

File: core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala (hunk 2 above)
Comment (RayRoestenburg) [reply core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala:22]: Do you mean DeploymentContext relies on strimzi? I'm not sure what you mean?

Comment (RayRoestenburg) [reply core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala:22]: The DeploymentContext is fine without strimzi, the user then just can't use the 'managed topic feature'.

File: core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Actions.scala (hunk 1 above)
Comment (RayRoestenburg) [reply core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Actions.scala]: Done.

File: core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala (hunk 2 above)
Comment (RayRoestenburg) [reply core/cloudflow-operator/src/test/scala/cloudflow/operator/TestDeploymentContext.scala:22]: (There is more work needed after this PR..)

**Pull request merged**